presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -965,10 +965,12 @@ private Table prepareTable(ConnectorSession session, ConnectorTableMetadata tabl
 
         validateColumns(hiveStorageFormat, columnHandles);
 
+        MetastoreContext metastoreContext = new MetastoreContext(session.getIdentity(), session.getQueryId(), session.getClientInfo(), session.getSource(), getMetastoreHeaders(session), isUserDefinedTypeEncodingEnabled(session), columnConverterProvider);
+
         Map<String, HiveColumnHandle> columnHandlesByName = Maps.uniqueIndex(columnHandles, HiveColumnHandle::getName);
         List<Column> partitionColumns = partitionedBy.stream()
                 .map(columnHandlesByName::get)
-                .map(column -> Column.partitionColumn(column.getName(), column.getHiveType(), column.getComment()))
+                .map(columnHandle -> columnHandle.toColumn(session, columnConverterProvider))
```

Comment:
We would assume the `columnConverterProvider` passed in is the one we want already. So get `columnConverterProvider` through `metastoreContext.getColumnConverterProvider()`.